lib/Twig/Environment.php

```diff
@@ -1512,12 +1530,22 @@ protected function initExtension(Twig_ExtensionInterface $extension)
 
         // operators
         if ($operators = $extension->getOperators()) {
-            if (2 !== count($operators)) {
-                throw new InvalidArgumentException(sprintf('"%s::getOperators()" does not return a valid operators array.', get_class($extension)));
+            if (!is_array($operators)) {
+                throw new InvalidArgumentException(sprintf(
+                    '"%s::getOperators()" must return an operators array, got "%s".',
+                    get_class($extension), is_object($operators) ? get_class($operators) : gettype($operators).(is_resource($operators) ? '' : '#'.$operators)
+                ));
             }
 
-            $this->unaryOperators = array_merge($this->unaryOperators, $operators[0]);
-            $this->binaryOperators = array_merge($this->binaryOperators, $operators[1]);
+            switch (count($operators)) {
+                case 1:
```

Comment:
there is 2 thing here:

- it is not related to registering operators without extension, so it should not be hidden in an unrelated PR (the PR description never talks about this change, and it is not covered by tests either)
- I would be against it, as it makes the API even harder to understand IMO